src/irmin-pack/unix/dispatcher.ml

```diff
@@ -117,8 +120,16 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
   let read_exn t ~off ~len ?volume_identifier buf =
     read_range_exn t ~off ~min_len:len ~max_len:len ?volume_identifier buf
 
+  let read_sparse_or_lower_exn t ~off ~max_len buf =
```

Comment:
WDYT about tucking this function into `read_seq_exn` like the other helper functions it has?

Reply:
Yeah so the plan was to share more logic with `read_range_exn` (and did so in the latest commit, not sure if it was really worth it)

Reply:
The code is nice and tidy now! :clap: